beacon_node/beacon_chain/src/attestation_rewards.rs

```diff
@@ -0,0 +1,185 @@
+use beacon_chain::{BeaconChain, BeaconChainTypes};
+use eth2::lighthouse::attestation_rewards::{IdealAttestationRewards, TotalAttestationRewards};
+use eth2::lighthouse::StandardAttestationRewards;
+use participation_cache::ParticipationCache;
+use safe_arith::SafeArith;
+use slog::{debug, Logger};
+use state_processing::{
+    common::altair::BaseRewardPerIncrement,
+    per_epoch_processing::altair::{participation_cache, rewards_and_penalties::get_flag_weight},
+};
+use std::{collections::HashMap, sync::Arc};
+use store::consts::altair::PARTICIPATION_FLAG_WEIGHTS;
+use types::consts::altair::WEIGHT_DENOMINATOR;
+
+use types::{Epoch, EthSpec};
+
+use crate::{beacon_chain, BeaconChainError};
+
+impl<T: BeaconChainTypes> BeaconChain<T> {
+    pub fn compute_attestation_rewards(
+        &self,
+        epoch: Epoch,
+        chain: Arc<BeaconChain<T>>,
+        mut validators: Vec<usize>,
+        log: Logger,
+    ) -> Result<StandardAttestationRewards, BeaconChainError> {
+        debug!(log, "computing attestation rewards"; "epoch" => epoch, "validator_count" => validators.len());
+
+        //--- Get state ---//
+        let spec = &chain.spec;
+
+        let execution_optimistic = chain.is_optimistic_or_invalid_head()?;
+
+        let state_slot = (epoch + 1).end_slot(T::EthSpec::slots_per_epoch());
+
+        let state_root = chain.state_root_at_slot(state_slot)?;
+
+        let state = chain.get_state(&state_root.unwrap(), Some(state_slot))?;
+
+        //--- Calculate ideal_rewards ---//
+        let participation_cache = ParticipationCache::new(state.as_ref().unwrap(), spec)?;
+
+        let previous_epoch = state.as_ref().unwrap().previous_epoch();
+
+        let mut ideal_rewards_hashmap = HashMap::new();
+
+        for flag_index in PARTICIPATION_FLAG_WEIGHTS {
+            let weight = get_flag_weight(flag_index.try_into().unwrap()).unwrap_or_default();
```

Comment:
The Rust compiler will tell you to do this, but `as usize` is preferable (safe and doesn't need unwrap). We basically want to avoid ever using `unwrap` or `expect`.

We should also return an error instead of using `unwrap_or_default()`